pkg/apis/componentconfig/v1alpha1/temp_v1a1_refactor.gogo

```diff
@@ -0,0 +1,548 @@
+  // containers in each pod will use.
+  PodInfraContainerImage string `json:"podInfraContainerImage"`
+}
+
+type DockerOptions struct {
+  // dockerEndpoint is the path to the docker endpoint to communicate with.
+  DockerEndpoint string `json:"dockerEndpoint"`
+  // enable gathering custom metrics.
+  EnableCustomMetrics bool `json:"enableCustomMetrics"` // for Docker
+}
+
+type RktOptions struct {
+  // rktPath is the  path of rkt binary. Leave empty to use the first rkt in
+  // $PATH.
+  RktPath string `json:"rktPath"`
+  // rktApiEndpoint is the endpoint of the rkt API service to communicate with.
+  RktAPIEndpoint string `json:"rktAPIEndpoint"`
+}
+
+/* END RUNTIME REMOVAL */
+
+type NodeRegistrationConfig struct {
+    // registerNode enables automatic registration with the apiserver.
+  RegisterNode *bool `json:"registerNode"`
+  // registerWithTaints are an array of taints to add to a node object when
+  // the kubelet registers itself. This only takes effect when registerNode
+  // is true and upon the initial registration of the node.
+  RegisterWithTaints []v1.Taint `json:"registerWithTaints"`
+  // nodeLabels to add when registering the node in the cluster.
+  NodeLabels map[string]string `json:"nodeLabels"`
+
+}
+
+type NodeStatusConfig struct {
+  // nodeStatusUpdateFrequency is the frequency that kubelet posts node
+  // status to master. Note: be cautious when changing the constant, it
+  // must work with nodeMonitorGracePeriod in nodecontroller.
+  NodeStatusUpdateFrequency metav1.Duration `json:"nodeStatusUpdateFrequency"`
+}
+
+type ClusterDNSConfig struct {
+  // clusterDNS is a list of IP address for the cluster DNS server.  If set,
+  // kubelet will configure all containers to use this for DNS resolution
+  // instead of the host's DNS servers
+  ClusterDNS []string `json:"clusterDNS"`
+  // clusterDomain is the DNS domain for this cluster. If set, kubelet will
+  // configure all containers to search this domain in addition to the
+  // host's search domains.
+  ClusterDomain string `json:"clusterDomain"`
+}
+
+type KubeAPIConfig struct {
+  /* requests to API server */
+  // contentType is contentType of requests sent to apiserver.
+  ContentType string `json:"contentType"`
+  // kubeAPIQPS is the QPS to use while talking with kubernetes apiserver
+  KubeAPIQPS *int32 `json:"kubeAPIQPS"`
+  // kubeAPIBurst is the burst to allow while talking with kubernetes
+  // apiserver
+  KubeAPIBurst int32 `json:"kubeAPIBurst"`
+  // syncFrequency is the max period between synchronizing running
+  // containers and config
+  SyncFrequency metav1.Duration `json:"syncFrequency"` // TODO(kc-refactor): This feels right here but take another look later.
+}
+
+type FeatureGates struct {
+  // featureGates is a string of comma-separated key=value pairs that describe feature
+  // gates for alpha/experimental features.
+  FeatureGates string `json:"featureGates,omitempty"`
```

Comment:
Yes, it does. We ought to put this somewhere central, because so many components use it. Where that central place should be, I've yet to really think about.